Submit speakers to airtable

tuttiq opened the pull request:
Resolves #20 

**Form info**

![Screen Shot 2020-07-05 at 5 06 37 PM](https://user-images.githubusercontent.com/1096046/86528413-769b5880-bee2-11ea-9854-5619c3096088.png)

**Request payload**

![Screen Shot 2020-07-05 at 5 09 41 PM](https://user-images.githubusercontent.com/1096046/86528415-7c913980-bee2-11ea-815d-9d095e11b7c9.png)

**Console output**

![Screen Shot 2020-07-05 at 5 08 12 PM](https://user-images.githubusercontent.com/1096046/86528418-861aa180-bee2-11ea-9a28-adb643649574.png)

**Airtable database**

![Screen Shot 2020-07-05 at 5 11 49 PM](https://user-images.githubusercontent.com/1096046/86528436-a34f7000-bee2-11ea-83e1-56260b79d511.png)

ann-kilzer commented on web/src/components/nomination/NominateForm.vue:
```diff
@@ -48,40 +59,91 @@ export default {
           company: '',
         },
-        speakerBio: '',
+        speaker_bio: '',
```
Are we moving to snake case because airtable expects it?

tuttiq replied:
Yeah, just on the data layer that interacts with the database 🙏

ann-kilzer commented on web/src/components/nomination/NominateForm.vue:
```diff
@@ -1,36 +1,46 @@
-    <v-btn color="primary">
+    <v-btn
+      color="primary"
+      @click="submit()"
```
A better pattern for forms is to make the button have the submit property and then the form catches the submit event. See an example on my branch: 
https://github.com/WWCodeTokyo/speak-her-db/pull/47/files
in `web/src/components/nomination/NominateForm.vue`

And [this article](https://alligator.io/vuejs/vue-form-handling/)

ann-kilzer commented on web/src/components/nomination/TopicsInput.vue:
```diff
@@ -1,14 +1,10 @@
 <template>
   <v-row dense>
     <v-col>
-      <v-autocomplete
```
Oh perhaps I misunderstood your comment this morning... were you talking about changing the airtable form, or changing this? Because the autocomplete with multiple chips could work better than splitting up a string for both the UI and ease of data entry.

tuttiq replied:
Ah, I agree, autocomplete provides a better UX. The problem is that Airtable does not allow to create extra options dynamically for "multiple string select" columns.

The solution I see is to extract `Topics` into a separate table (like we're doing with `Location`) and then handle loading existing options and creating new options on the fly all on the front-end. It's some level of extra work, but maybe worth it?

Can we do it on a separate PR though? (I can revert this back to autocomplete with limited options/no creation of new options if that's better for now)

ann-kilzer replied:
Yeah let's solve on a separate PR. It's subtle and we don't need topics right away anyway, we can collect people through the airtable form in the interim. 💪 👩

tuttiq commented:
Ok, should be good to go now @ann-kilzer ! I reverted the topics field to a limited multi-select for now, but added some prep on the component to fetch topics from the database later on.

Also the UI problem I was having fixed itself when I moved the Languages field near the photo URL for some reason lol

![select](https://user-images.githubusercontent.com/1096046/86530476-e6fea580-bef3-11ea-9b95-6eb76e835560.gif)

ann-kilzer commented on web/src/components/nomination/NominateForm.vue:
```diff
@@ -48,40 +83,96 @@ export default {
+
+      // Airtable expects an array of objects with the key `fields`
+      return [{ fields: payloadFields }];
+    },
```
Let's re-enable camelcase 🐫 after this method.

ann-kilzer commented on web/src/components/nomination/NominateForm.vue:
```diff
@@ -48,40 +83,96 @@ export default {
+      /* eslint-disable camelcase */
+
+      // copy the the fields that don't need parsing from the form object
+      const {
```
🤔 We have three different ways of moving items from our form Object into our payload object. It might be more readable to simply assign each one like so: 

```
payloadFields = {
       email := this.form.email,
        job_title: this.form.job.title,
        company: this.form.job.company,
        city: this.form.location.city,
        location_id: [this.form.location.prefecture],
      };
```

Especially the Object.keys part is confusing because sometimes we're looking at things with a prefix, and other times we're looking for things with a suffix. It's harder to read the code and ensure we're setting all the right values.

tuttiq replied:
Yeah, this parsing is a bit of an over-complication.... I was wondering, do you think it actually makes sense to just change the `form` object to match the payload final format and then just assign directly there when the inputs are changed? This way we can get rid of parsing altogether, I'm thinking it makes more sense and make it simpler? What do you think?

ann-kilzer replied:
I agree with the caveat that merging my validation PR will be a lot more complex. Can we build the payloadFileds and then as a future refactor, restructure the form?

ann-kilzer commented on web/src/components/nomination/NominateForm.vue:
```diff
@@ -48,40 +83,96 @@ export default {
+    },
+    afterSave(err, records) {
+      if (err) {
+        // Do something on error
```
nit: comments are overkill here :)

ann-kilzer commented on web/src/components/nomination/TopicsInput.vue:
```diff
@@ -27,16 +27,29 @@ export default {
+    // this.$getTopics(this.setTopics, this.setError);
+  },
   methods: {
+    setTopics(records) {
```
👍

ann-kilzer commented:
Oh also, we can do this as a follow-up, but we'll need to add a v-alert on successful upload and clear the form. Currently it appears it's not submitting, so I am clicking multiple times. I can make a new issue.
